### PR TITLE
refactor: store config query

### PR DESCRIPTION
**Pull request by bartoszherba**
## Description
- move loading of the store config to the plugin
- add store config documentation

## Motivation and Context
Perf improvement and better code base

## How Has This Been Tested?
-

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/plugins/storeConfigPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { PiniaPluginContext } from 'pinia';
+import { Plugin } from '@nuxt/types';
+import { ref, set } from '@nuxtjs/composition-api';
+import StoreConfigGql from '~/plugins/query/StoreConfig.gql';
+
+const storeConfigPlugin: Plugin = async ({ $pinia, $graphql }) => {
+  const configData = await $graphql.query.request(StoreConfigGql);
+  $pinia.use(({ store }: PiniaPluginContext) => {
+    if (store.$id !== 'magentoConfig') return;
+    // eslint-disable-next-line no-prototype-builtins
```
Comment (sethidden) [inline packages/theme/plugins/storeConfigPlugin.ts:10]:
```suggestion
```
There are no prototype builtins used on the following line, so this eslint magic comment doesn't do anything

File: packages/theme/components/StoreSwitcher/StoresModal.vue [hunk 2]
```diff
@@ -62,7 +62,10 @@ export default defineComponent({
   },
   props: {
     isLangModalOpen: Boolean,
-    storeConfig: Object,
+    storeConfig: {
+      type: Object as PropType<StoreConfig>,
+      default: () => {},
```
Comment (sethidden) [inline packages/theme/components/StoreSwitcher/StoresModal.vue:67]: eh we have the annoying eslint rule where it forces us to provide a default value

would it be possible to use like as `PropType<StoreCofnig | null>`? The `{}` doesn't really match StoreConfig

File: docs/guide/configuration.md [hunk 3]
```diff
@@ -45,6 +45,27 @@ Then on the `config` folder create a file `dev.json` with your configurations.
...
+        store_name,
+        default_display_currency_code,
+        locale,
+        header_logo_src,
+        logo_width,
+        logo_height,
+        logo_alt
+    }
+  }
+```
```
Comment (sethidden) [inline docs/guide/configuration.md:68]:
```suggestion
```
This is already in StoreConfig.gql.ts and the added docs contents already point to that file - do we want to maintain this here aswell?

**Review by Frodigo — DISMISSED** [review]

File: packages/theme/components/StoreSwitcher/StoresModal.vue (hunk 2 above)
Comment (bartoszherba) [reply packages/theme/components/StoreSwitcher/StoresModal.vue:67]: Maybe we should remove that rule then?

Comment (bartoszherba) [reply packages/theme/components/StoreSwitcher/StoresModal.vue:67]: Anyway, I updated the code.

Comment (sethidden) [reply packages/theme/components/StoreSwitcher/StoresModal.vue:67]: Did you push it out? I can't see the updates

**Review by sethidden — APPROVED** [review]

**Pull request merged**